src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -1973,9 +1973,9 @@ namespace winrt::TerminalApp::implementation
                 }
             }
 
-            std::wregex newLine{ LR"((\r|\n))" };
-            const bool hasNewLine = std::regex_search(text.c_str(), newLine);
-            const bool warnMultiLine = hasNewLine && _settings.GlobalSettings().WarnAboutMultiLinePaste();
+            const auto hasNewLineLambda = [&](auto c) { return c == L'\n' || c == L'\r'; };
```

Comment:
no need for capture here :smile:

```suggestion
            const auto hasNewLineLambda = [](auto c) { return c == L'\n' || c == L'\r'; };
```

nit: prefer name `isNewLineLambda` (you're finding it "if it is a newline")